beacon-chain/core/helpers/committee.go

```diff
@@ -504,11 +504,20 @@ func NextEpochSyncSubcommitteeIndices(
 
 // UpdateSyncCommitteeCache updates sync committee cache.
 func UpdateSyncCommitteeCache(state iface.BeaconStateAltair) error {
-	r, err := syncPeriodBoundaryRoot(state)
+	nextSlot := state.Slot() + 1
+	if nextSlot%params.BeaconConfig().SlotsPerEpoch != 0 {
+		return errors.New("not at the end of the epoch to update cache")
+	}
+	if SlotToEpoch(nextSlot)%params.BeaconConfig().EpochsPerSyncCommitteePeriod != 0 {
+		return errors.New("not at sync committee period boundary to update cache")
+	}
+
+	prevBlockRoot, err := state.LatestBlockHeader().HashTreeRoot()
```

Comment:
This is a bug, the latest block header always has its state root zeroed out. So when filling up the cache, the underlying root is a garbled block root. We could manually correct the block header's state root and then recompute the HTR.

Reply:
It's not a bug here. `UpdateSyncCommitteeCache` is used during `ProcessEpoch` which means it's used after `ProcessSlot` and before `ProcessBlock`.

I do see the concerns regarding miss usage and we can make this better. Let's return error if latest block header's state root is zero hash to prevent miss usages. We should also update the comments to reflect this

Reply:
Ok thanks, that would be more helpful here. Any new reader would be able to grasp the context better and not misuse this.